price_parser/parser.py

```diff
@@ -182,16 +186,16 @@ def extract_price_text(price: str) -> Optional[str]:
     """
     if price.count('€') == 1:
         m = re.search(r"""
-        [\d\s.,]*?\d    # number, probably with thousand separators
-        \s*?€\s*?        # euro, probably separated by whitespace
-        \d\d
-        (?:$|[^\d])    # something which is not a digit
+        [\d\s.,]*?\d    # number, probably with thousand separators                
+        \s*?€(\s*?)?    # euro, probably separated by whitespace
```

Comment:
It looks to me like you mean to capture the optional space after the € here in capture group 1, and then below you mean to match different content depending on whether this capture group matched content or not. But I’m not aware if this feature of Python regular expressions (`(?(N)A|B)` syntax).

To me it seems like `(\s*?)?\d(?(1)\d|\d*?)` is functionally equivalent to `\s*?\d\d*?`, which goes from only accepting 2 digits after €-as-a-decimal-separator to accepting 1 or more.

Reply:
In this step we don't know about of € is a decimal-separator or not, we are trying to extract price text. So, if we find a € in text, we are looking for whitespaces after  €, if we find one or more than we should pick it if we have only 2 digits, multiple digits othervise.
I'm not sure that ```(\s*?)?\d(?(1)\d|\d*?)``` is equivalent to ```\s*?\d\d*?```, because tests are not passed after replacing it.
I found a (?(id/name)yes-pattern|no-pattern) - from official re python lib documentation.

Reply:
I had no idea about `(?(id/name)yes-pattern|no-pattern)`, great to know :smiley: